### ISSUE TITLE
Convoke card cast with too much creatures on battlefield might crash the server

### ISSUE THREAD

**Issue by Zerrisx** [magefree/mage#6938]
I'm pretty sure I crashed both .de and .us today.  The first time I cast March of the Multitudes for X=100 (using #6937).  Immediately after the spell resolved, the .de server crashed (and has been down since).  I then tested on the .us server, and immediately after casting March for X=50, it also crashed, though it came back up a minute later.

In a different game on .us, casting for a mere X=30 did not crash.

Until further notice, though, I'll stop testing this on public servers...

**Comment by JayDi85** [comment]
I can't confirm it on local server. Maybe you have another cards on battlefield to multiply mana or effects like "add any color of available mana/permanents colors".

P.S. de restarted

**Comment by Zerrisx** [comment]
I also did not get it on my local server, though - if you're not worried - I can always try crashing a public server again...

**Comment by JayDi85** [comment]
Don't test crush bugs on public servers, only on beta server.

**Comment by JayDi85** [comment]
I found the real reason: convoke mana calculation generates too much data.

Use case:
```
[test server crush]
hand:Human:March of the Multitudes:1
battlefield:Human:Grizzly Bears:500
```

Error (with some additional logs):
```
INFO  2020-08-04 04:48:26,756 mana options size: 1                                                                       =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,757 mana options size: 2                                                                       =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,757 mana options size: 4                                                                       =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,757 mana options size: 8                                                                       =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,757 mana options size: 16                                                                      =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,757 mana options size: 32                                                                      =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,758 mana options size: 64                                                                      =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,758 mana options size: 128                                                                     =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,758 mana options size: 256                                                                     =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,758 mana options size: 512                                                                     =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,759 mana options size: 1024                                                                    =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,760 mana options size: 2048                                                                    =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,761 mana options size: 4096                                                                    =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,764 mana options size: 8192                                                                    =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,768 mana options size: 16384                                                                   =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,777 mana options size: 32768                                                                   =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,794 mana options size: 65536                                                                   =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,801 mana options size: 131072                                                                  =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,810 mana options size: 262144                                                                  =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,825 mana options size: 524288                                                                  =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:26,879 mana options size: 1048576                                                                 =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:27,543 mana options size: 2097152                                                                 =>[CALL main-42] ManaOptions.<init> 
INFO  2020-08-04 04:48:29,378 mana options size: 4194304                                                                 =>[CALL main-42] ManaOptions.<init> 
Exception in thread "CALL main-42" java.lang.OutOfMemoryError: GC overhead limit exceeded
	at mage.Mana.copy(Mana.java:584)
	at mage.abilities.mana.ManaOptions.<init>(ManaOptions.java:40)
	at mage.abilities.mana.ManaOptions.copy(ManaOptions.java:343)
	at mage.abilities.mana.ManaOptions.addMana(ManaOptions.java:328)
	at mage.abilities.keyword.ConvokeAbility.lambda$getManaOptions$1(ConvokeAbility.java:164)
	at mage.abilities.keyword.ConvokeAbility$$Lambda$122/1637368467.accept(Unknown Source)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.accept(ForEachOps.java:184)
	at java.util.stream.ReferencePipeline$2$1.accept(ReferencePipeline.java:175)
	at java.util.ArrayList$ArrayListSpliterator.forEachRemaining(ArrayList.java:1382)
	at java.util.stream.AbstractPipeline.copyInto(AbstractPipeline.java:481)
	at java.util.stream.AbstractPipeline.wrapAndCopyInto(AbstractPipeline.java:471)
	at java.util.stream.ForEachOps$ForEachOp.evaluateSequential(ForEachOps.java:151)
	at java.util.stream.ForEachOps$ForEachOp$OfRef.evaluateSequential(ForEachOps.java:174)
	at java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:234)
	at java.util.stream.ReferencePipeline.forEach(ReferencePipeline.java:418)
	at mage.abilities.keyword.ConvokeAbility.getManaOptions(ConvokeAbility.java:154)
	at mage.players.PlayerImpl.findActivatedAbilityFromPlayable(PlayerImpl.java:3239)
	at mage.players.PlayerImpl.getPlayable(PlayerImpl.java:3461)
	at mage.players.PlayerImpl.getPlayableObjects(PlayerImpl.java:3583)
	at mage.server.game.GameSessionPlayer.getGameView(GameSessionPlayer.java:192)
	at mage.server.game.GameSessionWatcher.lambda$update$0(GameSessionWatcher.java:52)
	at mage.server.game.GameSessionWatcher$$Lambda$90/318335075.accept(Unknown Source)
	at java.util.Optional.ifPresent(Optional.java:159)
	at mage.server.game.GameSessionWatcher.update(GameSessionWatcher.java:52)
	at mage.server.game.GameController.updateGame(GameController.java:758)
	at mage.server.game.GameController.cheat(GameController.java:679)
	at mage.server.game.GameManager.cheat(GameManager.java:133)
```

**Comment by JayDi85** [comment]
Fixed via 3430013f8dd0fec624a84df90c3520cd340fbc95
